### DISCUSSION TITLE
e2e chat and voice?

### DISCUSSION THREAD

**Discussion by vsatmydynipnet** [balzack/databag#108] [General]
I was reading through your project and was not able to find a answer to the question if voice is e2e too?

I assume text is not e2e by default, but you can seal the text chat.

Very interesting project.

**Comment by balzack** [comment]
Thanks for your interest! 

The voice chat is implemented using webrtc, which is a standard part of web browsers and a react-native module for mobile. I am not an authority on webrtc, but from my reading it is e2e encrypted as long as the source and destinations are the clients (which is the case with databag).

Some other implementations stream to a central server where the audio and video are composited, and sent as a single stream for the purposes of an efficient group conversation. That scenario would not qualify as e2e encrypted, but is also not how databag works.

So yes, databag voice is e2e encrypted, with the caveat that I am not an authority in webrtc.

To your last point, whether text is e2e encrypted by default (sealed), is now an admin setting. The person hosting the instance decides whether to allow unsealed conversations.

**Comment by vsatmydynipnet** [comment]
Thank you. This sounds really interesting. I will setup things for testing  here soon.
